guppylang/checker/core.py

```diff
@@ -109,6 +109,14 @@ def replace_defined_at(self, node: AstNode | None) -> "Variable":
         return replace(self, defined_at=node)
 
 
+@dataclass(frozen=True, kw_only=True)
+class ComptimeVariable(Variable):
```

Comment:
Following https://github.com/CQCL/guppylang/issues/898 we can probably merge this functionality into the regular `Variable`, but this seems ok for a short-term fix 👍